andela/Titan-Devs
# #163518696 Pagination support when fetching the articles

#### What does this PR do?
- It fetches all articles with pagination support
#### Description of Task to be completed?
- The user should be able to read articles created by other users with pagination support
#### How should this be manually tested?
- After cloning the repository, check out `ft-articles-pagination-endpoints-163518696` branch
- Make sure to read `README.md` file to get the setup instructions
- Using any `API client tool(usually POSTMAN)`,
- Create the user and login to get the `access-token`
- Create some articles using `POST /api/v1/articles` endpoint
- Make sure to pass the token under `request headers`. It should be {`Authorization`: `Bearer <access-token>`} KEY, VALUE PAIR
- the request body should be as mentioned in the `README.md` file under `Endpoints:` ==> `Create Article` subtitle
After the successful article(s) creation,
- You also can fetch all articles using `GET /api/v1/articles`

The `GET /api/v1/articles`  returns most recent articles globally. You can pass the `page=x` param where `x` is the page number(x=1 by default) and/or the `limit=y` where `y` is the maximum number of articles(y=20 by default) you want to get back.
 
#### Any background context you want to provide?
- The pagination context defined in the README.md file was to use `offset` query param in place of `page`, but I decided to use the `page` because it is obvious and makes sense than `offset`.
- This API endpoint can be accessed by any user and it doesn't require the user to be logged in
#### What are the relevant pivotal tracker stories?
- #163518696
